python/rascaline/tests/utils/splines.py

```diff
@@ -326,3 +326,48 @@ def test_center_contribution_gto_gaussian():
     center_contr_analytical *= normalization * 2 * np.pi / np.sqrt(4 * np.pi)
 
     assert_allclose(spliner.center_contribution, center_contr_analytical, rtol=1e-14)
+
+
+def test_custom_radial_integral():
+    cutoff = 2.0
+    max_radial = 3
+    max_angular = 2
+    atomic_gaussian_width = 1.2
+    n_spline_points = 20
+
+    spliner_args = {
+        "max_radial": max_radial,
+        "max_angular": max_angular,
+        "cutoff": cutoff,
+        "basis": GtoBasis(cutoff=cutoff, max_radial=max_radial),
+    }
+
+    spliner_analytical = SoapSpliner(
+        density=GaussianDensity(atomic_gaussian_width),
+        **spliner_args,
+    )
+
+    # Create a custom density that has not type "GaussianDensity" to trigger full
+    # numerical evaluation of the radial integral.
+    class mydensity(GaussianDensity):
+        pass
+
+    spliner_numerical = SoapSpliner(
+        density=mydensity(atomic_gaussian_width),
+        **spliner_args,
+    )
+
+    splines_analytic = spliner_analytical.compute(n_spline_points)
+    splines_numerical = spliner_numerical.compute(n_spline_points)
+
+    n_points = len(splines_analytic["TabulatedRadialIntegral"]["points"])
+
+    # TODO fix 0 component!!
```

Comment:
@kvhuguenin also this puzzles me a bit. Everything is fine, but the spline `derivatives` for r=0 are not very accurate for some values.